client/jetpack-cloud/sections/agency-dashboard/sites-overview/index.tsx

```diff
@@ -191,6 +191,35 @@ export default function SitesOverview() {
 		} );
 	}, [ selectedLicensesSiteId, selectedLicenses ] );
 
+	const renderAddSiteIssueLicenseButtons = () => {
+		return (
+			<div className="sites-overview__add-site-issue-license-buttons">
+				<Button
+					className="sites-overview__issue-license-button"
+					href="/partner-portal/issue-license"
+					onClick={ () =>
+						dispatch(
+							recordTracksEvent( 'calypso_jetpack_agency_dashboard_issue_license_button_click' )
+						)
+					}
+				>
+					{ translate( 'Issue License', { context: 'button label' } ) }
+				</Button>
+				<Button
+					primary
+					href="https://wordpress.com/jetpack/connect"
+					onClick={ () =>
+						dispatch(
+							recordTracksEvent( 'calypso_jetpack_agency_dashboard_add_site_button_click' )
+						)
+					}
+				>
+					{ translate( 'Add New Site', { context: 'button label' } ) }
+				</Button>
+			</div>
+		);
+	};
```

Comment:
Non-blocking, but recommended: One thing I try to do in situations like this, is to create small additional components inside the same file, rather than build `renderThing` functions, just to keep the logic of the main component clear and simple. I realize the render function "pattern" is already present in this file, though, so I totally understand.

The alternative I'm proposing could look something like this:

```tsx
const AddSiteIssueLicenseButtons = () => {
	const dispatch = useDispatch();
	const translate = useTranslate();

	return (
			<div className="sites-overview__add-site-issue-license-buttons">
				<Button
					className="sites-overview__issue-license-button"
					href="/partner-portal/issue-license"
					onClick={ () =>
						dispatch(
							recordTracksEvent( 'calypso_jetpack_agency_dashboard_issue_license_button_click' )
						)
					}
				>
					{ translate( 'Issue License', { context: 'button label' } ) }
				</Button>
				<Button
					primary
					href="https://wordpress.com/jetpack/connect"
					onClick={ () =>
						dispatch(
							recordTracksEvent( 'calypso_jetpack_agency_dashboard_add_site_button_click' )
						)
					}
				>
					{ translate( 'Add New Site', { context: 'button label' } ) }
				</Button>
			</div>
		);
}
```

... and then the references would be replaced like so:

```tsx
<SiteContentHeader
	content={
		isLargeScreen &&
		( selectedLicensesCount > 0
			? renderIssueLicenseButton()
			: <AddSiteIssueLicenseButtons /> )
	}
```

```tsx
{ ! isLargeScreen && <AddSiteIssueLicenseButtons /> }
```

Reply:
That's a good idea, and generally how I do that too, but I'm always a bit shy to do things completely differently when working with existing code like that.